packages/server/graphql/mutations/helpers/bootstrapNewUser.ts

```diff
@@ -30,10 +28,9 @@ const bootstrapNewUser = async (
   const {id: userId, createdAt, preferredName, email, featureFlags, tier, segmentId} = newUser
   // email is checked by the caller
   const domain = email.split('@')[1]!
-  const [isPatient0, usersWithDomain] = await Promise.all([
-    isPatientZero(domain),
-    isCompanyDomain(domain) ? getUsersbyDomain(domain) : []
-  ])
+  const isCompanyDomain = await dataLoader.get('isCompanyDomain').load(domain)
+  const usersWithDomain = isCompanyDomain ? await getUsersbyDomain(domain) : []
+  const isPatient0 = !!domain && isCompanyDomain && usersWithDomain.length === 0
```

Comment:
I inlined the `isPatientZero` call as this is the only usage and it was doing a second uncached roundtrip to PG to check whether there are other users with the same domain. These users are already loaded in `usersWithDomain`.

Reply:
Easiest to test this is to add some debug output here